pkg/controller/handlers/mcpserver/mcpserver.go

```diff
@@ -0,0 +1,71 @@
+package mcpserver
+
+import (
+	"net/url"
+	"slices"
+
+	"github.com/obot-platform/nah/pkg/router"
+	"github.com/obot-platform/obot/apiclient/types"
+	v1 "github.com/obot-platform/obot/pkg/storage/apis/obot.obot.ai/v1"
+	"github.com/obot-platform/obot/pkg/utils"
+)
+
+func DetectDrift(req router.Request, _ router.Response) error {
+	server := req.Object.(*v1.MCPServer)
+
+	if server.Spec.MCPServerCatalogEntryName == "" {
+		return nil
+	}
+
+	var entry v1.MCPServerCatalogEntry
+	if err := req.Get(&entry, server.Namespace, server.Spec.MCPServerCatalogEntryName); err != nil {
+		return err
+	}
+
+	var (
+		drifted bool
+		err     error
+	)
+	if entry.Spec.CommandManifest.Name != "" {
+		drifted, err = configurationHasDrifted(server.Spec.Manifest, entry.Spec.CommandManifest)
+	} else {
+		drifted, err = configurationHasDrifted(server.Spec.Manifest, entry.Spec.URLManifest)
+	}
+	if err != nil {
+		return err
+	}
+
+	server.Status.NeedsUpdate = drifted
+	return req.Client.Status().Update(req.Ctx, server)
```

Comment:
Two comment here:
1. I believe that the Update call here is unnecessary because the status of the object will get persisted after all handlers are run.
2. If you do want to make this Update call, then you should only do it if it is necessary.
```suggestion
	if server.Status.NeedsUpdate != drifted {
		server.Status.NeedsUpdate = drifted
		return req.Client.Status().Update(req.Ctx, server)
	}
	return nil
```

Reply:
Fixed (decided to use your suggested code)